microsoft/PlanetaryComputerDataCatalog
# Layout cleanup

## Overview

This PR does a few things in service of cleanup and consistency:
1) Cleans up the `Header` component. b407625 and d6ca52d
2) Moves the overall layout to flex-box all the way down 249510b

### Header changes
1) Moves to a directory based approach for the component
2) Creates `styles.ts` to handle the quantity of `mergeStyles` generated classes
3) Moves `HeaderLink` into `src/components/Header/components` to co-locate the components that are only used within the Header component (this is a pretty opinionated move and the component could stay in `src/components` or even next to the index file as `src/components/Header/HeaderLink.tsx`)
4) Removes all inline styles and implements the _mostly_ same files using `mergeStyles` to generate dynamic CSS classes.
5) Makes a slight adjustment to yml import to please the TypeScript compiler
6) Converts any touched files to TS
7) Adjusts styles so when wrapping occurs, all header content is aligned to the same left axis

### Layout changes
The goal of these changes was to remove the need to have short and long layout variations along with the hardcoding or detection of header and footer sizes. For pages that might be shorter than the window height, the footer remains at the bottom of the window. When those pages grow, the footer appears at the bottom of the content as expected. For Fixed height layouts (such as the Explorer), we lean on flex-box to allow the content to grow to fill the screen. To summarize, this commit:

1) removes all `isShort` usage
2) removes all hardcoded sizes
3) adjusts the structure of the DOM and top-level CSS to allow for flex-box element stacking

## Testing
1) Ensure the header looks correct under various screen size scenarios
2) Ensure that the all other pages look correct in their overall layout, especially the Explorer.
